cmd/genesis-check/main.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 The Cube Authors
+// This file is part of the Cube library.
+//
+// The Cube library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Cube library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Cube library. If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/ethereum/go-ethereum/core"
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintln(os.Stderr, "Usage:", os.Args[0], "<genesis_file>")
+		flag.PrintDefaults()
+		fmt.Fprintln(os.Stderr, `Parses the given genesis file and tries to generate genesis block hash.`)
+	}
+}
+
+// checkGenesis uses the genesis file to generate genesis block
+func checkGenesis(file *os.File) {
+	genesis := new(core.Genesis)
+	if err := json.NewDecoder(file).Decode(genesis); err != nil {
+		die(err)
+	}
+	genesisHash := genesis.ToBlock(nil).Hash()
+	fmt.Printf("Genesis Hash: %v\n", genesisHash)
```

Comment:
May be it's better to add an additional check whether this `genesisHash` is matching the `params.MainnetGenesisHash` or `params.TestnetGenesisHash`.

Reply:
Done